src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -92,7 +92,7 @@ class IqPolicyEvaluatorUtil
       throw e
     }
     else {
-      listener.logger.println Messages.IqPolicyEvaluation_UnableToCommunicate(e.message)
+      listener.logger.println Messages.IqPolicyEvaluation_Failed(e.message, e.cause?.message)
```

Comment:
Instead of trying to handle two error message in one, I would use two error messages. One for the main exception, the second for the "cause" exception. Something that would generate something like this in the log:
```
IQ Server request failed because: Something went wrong
Caused by: Something else went wrong
```

On the other hand, this kind of problem has a better solution in Java because Java exceptions have a stack trace and that is how you usually get all the info about the error, including all the messages in the exception chain. After all how do you know that you only want the first level "cause" and that there aren't other levels that are important?

So, why not log the exception itself with all the stack trace?

Reply:
>So, why not log the exception itself with all the stack trace?

That's a good question.  In Bamboo the full stack would show up, and in any other clients the full stack would show up.  Wonder if we can just print the entire exception.

Reply:
How about we remove the try catch.  With it gone the following is shown.  

```
Started by user anonymous
Building in workspace /Users/jameswilcox/clm-9736-branch-copy/nexus-platform-plugin/work/workspace/test-free
FATAL: Could not verify application public id from IQ Server
com.sonatype.nexus.api.exception.IqClientException: Could not verify application public id from IQ Server
	at com.sonatype.nexus.api.cy.verifyOrCreateApplication(SourceFile:78)
	at com.sonatype.nexus.api.iq.IqClient$verifyOrCreateApplication.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorUtil.evaluatePolicy(IqPolicyEvaluatorUtil.groovy:52)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorUtil$evaluatePolicy.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorBuildStep.perform(IqPolicyEvaluatorBuildStep.groovy:77)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: org.apache.http.client.HttpResponseException: Public ID cannot contain whitespaces.
	at com.sonatype.nexus.api.m.a(SourceFile:60)
	at com.sonatype.nexus.api.m.a(SourceFile:65)
	at com.sonatype.nexus.api.n.a(SourceFile:67)
	at com.sonatype.nexus.api.cy.verifyOrCreateApplication(SourceFile:75)
	... 19 more
Finished: FAILURE
```

Reply:
> How about we remove the try catch. With it gone the following is shown.

Note that removing the try-catch does end up changing the Jenkins behavior as well, i.e. we get a status of `FAILURE` rather than `UNSTABLE`. 

The funny thing is, for a situation where the application ID is invalid, then `FAILURE` is likely the right result. But for other network errors, things where it could have just been a blink in connectivity or a server temporarily being down, `UNSTABLE` arguably seems more justifiable.

I think it really comes down to how drastic or invasive we want to be in terms of making this change. There may be a good reason for the current behavior, to be honest, even if it's not the most intuitive for this new use case.

Reply:
> FAILURE rather than UNSTABLE

How about f597b0f

It produces:

```
Started by user anonymous
Building in workspace /Users/jameswilcox/clm-9736-branch-copy/nexus-platform-plugin/work/workspace/test-free
ERROR: Could not verify application public id from IQ Server
com.sonatype.nexus.api.exception.IqClientException: Could not verify application public id from IQ Server
	at com.sonatype.nexus.api.cy.verifyOrCreateApplication(SourceFile:78)
	at com.sonatype.nexus.api.iq.IqClient$verifyOrCreateApplication.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorUtil.evaluatePolicy(IqPolicyEvaluatorUtil.groovy:52)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorUtil$evaluatePolicy.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.sonatype.nexus.ci.iq.IqPolicyEvaluatorBuildStep.perform(IqPolicyEvaluatorBuildStep.groovy:77)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: org.apache.http.client.HttpResponseException: Public ID cannot contain whitespaces.
	at com.sonatype.nexus.api.m.a(SourceFile:60)
	at com.sonatype.nexus.api.m.a(SourceFile:65)
	at com.sonatype.nexus.api.n.a(SourceFile:67)
	at com.sonatype.nexus.api.cy.verifyOrCreateApplication(SourceFile:75)
	... 19 more
Build step 'Invoke Nexus Policy Evaluation' changed build result to UNSTABLE
Finished: UNSTABLE
```

Reply:
> How about f597b0f

Looks good to me as well, do the unit tests need to be updated for the new content? (I'm not sure the strings it's currently searching for will be there.)

Reply:
> do the unit tests need to be updated for the new content?
yes the tests need to updated

Reply:
> do the unit tests need to be updated for the new content?

Tests updated 3a2aa47